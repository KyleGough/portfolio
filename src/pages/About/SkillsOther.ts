import { Confidence,ISkill } from '../../components/SkillList';
import BashLogo from '../../img/bash.png';
import GitLogo from '../../img/git.png';
import LatexLogo from '../../img/latex.png';
import MarkdownLogo from '../../img/markdown.png';
import SQLLogo from '../../img/sql.png';
import TerminalLogo from '../../img/terminal.png';

export const skillsOther: ISkill[] = [
  {
    name: 'Git',
    progress: 85,
    className: 'w-[85%]',
    description: 'Used in all projects',
    confidence: Confidence.COMFORTABLE,
    logo: GitLogo,
  },
  {
    name: 'Bash',
    progress: 75,
    className: 'w-[75%]',
    description: 'Shell Scripting Exercises, Finding security defects in a VM',
    confidence: Confidence.COMFORTABLE,
    logo: BashLogo,
  },
  {
    name: 'Zsh',
    progress: 85,
    className: 'w-[85%]',
    description: 'Personal Shell of Choice, Everyday use with Manjaro i3',
    confidence: Confidence.COMFORTABLE,
    logo: TerminalLogo,
  },
  {
    name: 'SQL',
    progress: 60,
    className: 'w-[60%]',
    description:
      'Delivery Route Planner, BSplit, Department Store Database Analysis',
    confidence: Confidence.COMFORTABLE,
    logo: SQLLogo,
  },
  {
    name: 'LaTeX',
    progress: 50,
    className: 'w-[50%]',
    description: 'Dissertation, Multiple Group Projects ',
    confidence: Confidence.COMFORTABLE,
    logo: LatexLogo,
  },
  {
    name: 'Markdown',
    progress: 80,
    className: 'w-[80%]',
    description: 'Lecture Notes, GitHub READMEs',
    confidence: Confidence.CONFIDENT,
    logo: MarkdownLogo,
  },
];
