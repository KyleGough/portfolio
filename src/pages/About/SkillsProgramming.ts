import { ISkill, Confidence } from '../../components/SkillList';
import PythonLogo from '../../img/python.png';
import JavaLogo from '../../img/java.png';
import CSharpLogo from '../../img/c#.png';
import CPlusPlusLogo from '../../img/c++.png';
import VBLogo from '../../img/vb.png';
import HaskellLogo from '../../img/haskell.png';
import RubyLogo from '../../img/ruby.png';
import MatlabLogo from '../../img/matlab.png';

export const skillsProgramming: ISkill[] = [
  {
    name: 'Python',
    progress: 90,
    className: 'w-[90%]',
    description: 'Logical Sudoku Solver, Machine Learning, Image Analysis',
    confidence: Confidence.CONFIDENT,
    logo: PythonLogo,
  },
  {
    name: 'Java',
    progress: 80,
    className: 'w-[80%]',
    description: 'Robot Maze Environment, Steganography, Witter',
    confidence: Confidence.CONFIDENT,
    logo: JavaLogo,
  },
  {
    name: 'C#',
    progress: 75,
    className: 'w-[75%]',
    description: 'Sorting Visualiser, Delivery Route Planner',
    confidence: Confidence.CONFIDENT,
    logo: CSharpLogo,
  },
  {
    name: 'C++',
    progress: 65,
    className: 'w-[65%]',
    description: 'GLUT Simulation of a Roller Coaster',
    confidence: Confidence.COMFORTABLE,
    logo: CPlusPlusLogo,
  },
  {
    name: 'Visual Basic',
    progress: 50,
    className: 'w-[50%]',
    description: 'First Programming Language, CLI Sorting Algorithm Visualiser',
    confidence: Confidence.COMFORTABLE,
    logo: VBLogo,
  },
  {
    name: 'Haskell',
    progress: 40,
    className: 'w-[40%]',
    description: 'Mastermind Coursework, Scratch Clone Coursework',
    confidence: Confidence.COMFORTABLE,
    logo: HaskellLogo,
  },
  {
    name: 'Ruby',
    progress: 30,
    className: 'w-[30%]',
    description: 'Minesweeper Game',
    confidence: Confidence.BEGINNER,
    logo: RubyLogo,
  },
  {
    name: 'MATLAB',
    progress: 45,
    className: 'w-[45%]',
    description: 'Digital Forensics on Images',
    confidence: Confidence.BEGINNER,
    logo: MatlabLogo,
  },
];
