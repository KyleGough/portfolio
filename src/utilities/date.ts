import dayjs from 'dayjs';

import { IDate, IProjectDate } from '../data';

export const getFormattedDate = (date: IDate) => {
  return dayjs()
    .month(date.month - 1)
    .year(date.year)
    .format('MMMM YYYY');
};

export const getShortDate = (date: IDate) => {
  return dayjs()
    .month(date.month - 1)
    .year(date.year)
    .format('YYYY-MM');
};

export const getDateRange = (dt: IProjectDate) => {
  const start = getFormattedDate(dt.start);
  return dt.end ? start + ' - ' + getFormattedDate(dt.end) : start;
};
