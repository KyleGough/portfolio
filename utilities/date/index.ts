export { getDateRange, getFormattedDate, getShortDate } from './date';
