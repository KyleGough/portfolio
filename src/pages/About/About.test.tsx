import { render } from '@testing-library/react';
import React from 'react';

import { mockIntersectionObserver } from '../../utilities/mockIntersectionObserver';
import { About } from './About';

describe('About page', () => {
  beforeEach(mockIntersectionObserver);

  it('renders', async () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });
});
