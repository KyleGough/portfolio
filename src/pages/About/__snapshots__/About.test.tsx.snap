// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title mb-8 text-center md:text-left"
    >
      About Me
    </h1>
    <p
      class="mt-4 mb-8 max-w-reading"
    >
      Hello, I'm Kyle, a Front-End Engineer at Atom Learning and Computer Science MEng graduate of the University of Warwick. Interests lie in bouldering, cycling, guitar, movies, and physics. I enjoy learning new languages, frameworks and technologies and have pursued multiple personal projects putting these skills into practice, detailed below.
    </p>
    <a
      class="w-fit shimmer group flex items-center px-12 py-4 bg-background text-link hover:text-link-hover focus:text-link-hover rounded-full border-2 shadow border-link hover:border-link-hover focus:border-link-hover"
      href="/CV.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p>
        Read my CV
      </p>
      <svg
        class="float-right ml-4 w-4 h-4 fill-link group-hover:fill-link-hover group-focus:fill-link-hover"
        enable-background="new 0 0 24 24"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0,0h24v24H0V0z"
            fill="none"
          />
        </g>
        <g>
          <polygon
            points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
          />
        </g>
      </svg>
    </a>
  </div>
  <hr
    class="mt-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
    id="programming-languages"
  >
    <h2
      class="project-header text-center md:text-left"
    >
      Programming Languages
    </h2>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[90%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 90;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Python logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="python.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Python
            </span>
             -
             
            <span
              class="text-sm"
            >
              Logical Sudoku Solver, Machine Learning, Image Analysis
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[80%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 80;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Java logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="java.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Java
            </span>
             -
             
            <span
              class="text-sm"
            >
              Robot Maze Environment, Steganography, Witter
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[75%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 75;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="C# logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="c#.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              C#
            </span>
             -
             
            <span
              class="text-sm"
            >
              Sorting Visualiser, Delivery Route Planner
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[65%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 65;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="C++ logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="c++.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              C++
            </span>
             -
             
            <span
              class="text-sm"
            >
              GLUT Simulation of a Roller Coaster
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[50%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 50;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Visual Basic logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="vb.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Visual Basic
            </span>
             -
             
            <span
              class="text-sm"
            >
              First Programming Language, CLI Sorting Algorithm Visualiser
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[40%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 40;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Haskell logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="haskell.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Haskell
            </span>
             -
             
            <span
              class="text-sm"
            >
              Mastermind Coursework, Scratch Clone Coursework
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[30%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 30;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Ruby logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="ruby.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Ruby
            </span>
             -
             
            <span
              class="text-sm"
            >
              Minesweeper Game
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Beginner
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-programming progress overflow-hidden"
        >
          <div
            class="w-[45%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 45;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="MATLAB logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="matlab.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              MATLAB
            </span>
             -
             
            <span
              class="text-sm"
            >
              Digital Forensics on Images
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Beginner
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
    id="web-development"
  >
    <h2
      class="project-header text-center md:text-left"
    >
      Web Development
    </h2>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[90%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 90;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="JavaScript logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="javascript.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              JavaScript
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio, LucidLab, Graph Algorithm Visualiser, RSCBot
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[70%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 70;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="TypeScript logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="typescript.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              TypeScript
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[75%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 75;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="React logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="react.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              React
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio, LucidLab, Minesweeper
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[70%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 70;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Tailwind logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="tailwind.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Tailwind
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[90%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 90;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="HTML logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="html.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              HTML
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio, BSplit, To Do List
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[90%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 90;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="CSS logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="css.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              CSS
            </span>
             -
             
            <span
              class="text-sm"
            >
              Portfolio, LucidLab, BSplit, Minesweeper
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-web progress overflow-hidden"
        >
          <div
            class="w-[55%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 55;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="PHP logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="php.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              PHP
            </span>
             -
             
            <span
              class="text-sm"
            >
              BSplit, Portfolio
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <p
      class="font-extrabold text-lg text-center md:text-left mb-8 mt-16"
    >
      Additional experience in
    </p>
    <div
      class="grid xl:grid-cols-12 md:grid-cols-6 grid-cols-3"
    >
      <div
        class="text-center mb-8"
      >
        <img
          alt="Node.js Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="node.png"
        />
        <p
          class="text-sm"
        >
          Node.js
        </p>
      </div>
      <div
        class="text-center mb-8"
      >
        <img
          alt="jQuery Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="jquery.png"
        />
        <p
          class="text-sm"
        >
          jQuery
        </p>
      </div>
      <div
        class="text-center mb-8"
      >
        <img
          alt="Bootstrap Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="bootstrap.png"
        />
        <p
          class="text-sm"
        >
          Bootstrap
        </p>
      </div>
      <div
        class="text-center mb-8"
      >
        <img
          alt="Materialize Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="materialize.png"
        />
        <p
          class="text-sm"
        >
          Materialize
        </p>
      </div>
      <div
        class="text-center mb-8"
      >
        <img
          alt="Webix Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="webix.png"
        />
        <p
          class="text-sm"
        >
          Webix
        </p>
      </div>
      <div
        class="text-center mb-8"
      >
        <img
          alt="Heroku Logo"
          class="w-8 h-8 mb-2 mx-auto"
          loading="lazy"
          src="heroku.png"
        />
        <p
          class="text-sm"
        >
          Heroku
        </p>
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
    id="other-technologies"
  >
    <h2
      class="project-header text-center md:text-left"
    >
      Other Technologies
    </h2>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[85%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 85;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Git logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="git.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Git
            </span>
             -
             
            <span
              class="text-sm"
            >
              Used in all projects
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[75%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 75;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Bash logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="bash.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Bash
            </span>
             -
             
            <span
              class="text-sm"
            >
              Shell Scripting Exercises, Finding security defects in a VM
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[85%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 85;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Zsh logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="terminal.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Zsh
            </span>
             -
             
            <span
              class="text-sm"
            >
              Personal Shell of Choice, Everyday use with Manjaro i3
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[60%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 60;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="SQL logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="sql.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              SQL
            </span>
             -
             
            <span
              class="text-sm"
            >
              Delivery Route Planner, BSplit, Department Store Database Analysis
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[50%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 50;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="LaTeX logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="latex.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              LaTeX
            </span>
             -
             
            <span
              class="text-sm"
            >
              Dissertation, Multiple Group Projects 
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Comfortable
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-center py-4"
    >
      <div
        class="w-3/12 lg:w-4/12 pt-2"
      >
        <div
          class="progress-other progress overflow-hidden"
        >
          <div
            class="w-[80%] custom-duration progress-bar transition-transform ease-in-out -translate-x-full"
            style="--duration-factor: 80;"
          />
        </div>
      </div>
      <div
        class="w-9/12 lg:w-8/12 ml-6 flex"
      >
        <img
          alt="Markdown logo"
          class="opacity-0 w-8 h-8 mr-4 transition-opacity duration-500"
          loading="lazy"
          src="markdown.png"
        />
        <div>
          <p>
            <span
              class="font-extrabold"
            >
              Markdown
            </span>
             -
             
            <span
              class="text-sm"
            >
              Lecture Notes, GitHub READMEs
            </span>
          </p>
          <p
            class="text-sm text-link-hover mt-1"
          >
            Confident
          </p>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
    id="goals"
  >
    <h2
      class="project-header text-center md:text-left"
    >
      Goals for 2022
    </h2>
    <p
      class="my-4"
    >
      Here are some languages and technologies I am keen to learn in 2022.
    </p>
    <ul
      class="list-none mt-8 ml-14"
    >
      <li
        class="before:border-traffic-green relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        TypeScript
      </li>
      <li
        class="before:border-traffic-green relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        ESLint
      </li>
      <li
        class="before:border-traffic-amber relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        Extension API
      </li>
      <li
        class="before:border-traffic-red relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        Next.js
      </li>
      <li
        class="before:border-traffic-red relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        Web Sockets
      </li>
      <li
        class="before:border-traffic-red relative my-4 before:content-[\\"\\"] before:absolute before:-left-12 before:w-6 before:h-6 before:border-2 before:rounded-xl"
      >
        Docker
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
    id="university-modules"
  >
    <h2
      class="project-header text-center md:text-left"
    >
      University Modules
    </h2>
    <div
      class="grid grid-cols-1 md:grid-cols-2"
    >
      <div
        class="flex flex-row align-middle my-8"
      >
        <div
          class="flex items-center justify-center w-12"
        >
          <h3
            class="text-xl text-center -rotate-90 whitespace-nowrap"
          >
            Year 1 Modules
          </h3>
        </div>
        <ul
          class="list-none pl-4 py-4 border-l-2 rounded-xl"
        >
          <li>
            Programming for Computer Scientists
          </li>
          <li>
            Design of Information Structures
          </li>
          <li>
            Mathematics for Computer Scientists I
          </li>
          <li>
            Mathematics for Computer Scientists II
          </li>
          <li>
            Computer Organisation and Architecture
          </li>
          <li>
            Professional Skills
          </li>
          <li>
            Web Development Technologies
          </li>
          <li>
            Computer Security
          </li>
        </ul>
      </div>
      <div
        class="flex flex-row align-middle my-8"
      >
        <div
          class="flex items-center justify-center w-12"
        >
          <h3
            class="text-xl text-center -rotate-90 whitespace-nowrap"
          >
            Year 2 Modules
          </h3>
        </div>
        <ul
          class="list-none pl-4 py-4 border-l-2 rounded-xl"
        >
          <li>
            Operating Systems and Computer Networks
          </li>
          <li>
            Functional Programming
          </li>
          <li>
            Database Systems
          </li>
          <li>
            Formal Languages
          </li>
          <li>
            Algorithms
          </li>
          <li>
            Software Engineering
          </li>
          <li>
            Logic and Verification
          </li>
          <li>
            Cyber Security
          </li>
        </ul>
      </div>
      <div
        class="flex flex-row align-middle my-8"
      >
        <div
          class="flex items-center justify-center w-12"
        >
          <h3
            class="text-xl text-center -rotate-90 whitespace-nowrap"
          >
            Year 3 Modules
          </h3>
        </div>
        <ul
          class="list-none pl-4 py-4 border-l-2 rounded-xl"
        >
          <li>
            Dissertation
          </li>
          <li>
            Mobile Robotics
          </li>
          <li>
            Computer Graphics
          </li>
          <li>
            Neural Computing
          </li>
          <li>
            Machine Learning
          </li>
          <li>
            Project Management
          </li>
          <li>
            Digital Forensics
          </li>
        </ul>
      </div>
      <div
        class="flex flex-row align-middle my-8"
      >
        <div
          class="flex items-center justify-center w-12"
        >
          <h3
            class="text-xl text-center -rotate-90 whitespace-nowrap"
          >
            Year 4 Modules
          </h3>
        </div>
        <ul
          class="list-none pl-4 py-4 border-l-2 rounded-xl"
        >
          <li>
            High Peformance Computing
          </li>
          <li>
            Group Project
          </li>
          <li>
            Image and Video Analysis
          </li>
          <li>
            Quantum Computing
          </li>
          <li>
            Computational Biology
          </li>
          <li>
            Foundations of Data Analytics
          </li>
          <li>
            Advanced Computer Security
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
